pkg/mimirtool/client/backfill.go

```diff
@@ -100,11 +100,20 @@ func (c *MimirClient) backfillBlock(ctx context.Context, blockDir string, logctx
 		}
 	}
 
-	resp, err = c.doRequest(ctx, path.Join(endpointPrefix, url.PathEscape(blockID), finishBlockUpload), http.MethodPost, nil, -1)
-	if err != nil {
-		return errors.Wrap(err, "request to finish block upload failed")
+	for {
+		resp, err = c.doRequest(ctx, path.Join(endpointPrefix, url.PathEscape(blockID), finishBlockUpload), http.MethodPost, nil, -1)
+		if err != nil {
+			if errors.Is(err, errTooManyRequests) {
+				// wait and try again
+				logctx.WithField("error", err).Warning("will sleep and try again")
+				time.Sleep(sleepTime)
```

Comment:
dskit has a backoff mechanism [here](https://github.com/grafana/dskit/blob/727cbc22c1febfb5a77d024e407ebd01b207f74d/backoff/backoff.go#L26) that might be nice to use rather than directly sleeping.